### ISSUE TITLE
Using arithmetic operations in TypeQL queries

### ISSUE THREAD

**Issue by haikalpribadi** [vaticle/typedb#5521]
## Problem to Solve

Given the following `match` clause:
```
match $x isa declaration, has work-expense $we, has education-expense $ee;
```

We would like to perform arithmetic operations on `$we` and `$ee`, such as: 
```
$e = $we + $ee;
```

The arithmetic operations we would like to perform are:
- addition
- subtraction
- multiplication
- division

## Current Workaround

At the moment, users have to compute these operations manually and store them into the database.

## Proposed Solution

We introduce a special type of `statement` to perform arithmetic operations; let's say _"functional statements"_. This would allow us to write `pattern`s (in `match` clauses and `rule` bodies) such as:
```
$x isa declaration, has work-expense $we, has education-expense $ee;
$e = $we + $ee;
```

## Additional Information

As a first iteration of the solution, we can prohibit variables that contain results of arithmetic operations to be used further for comparisons.

**Comment by maydanw** [comment]
It may be that a sum(), sub(),.. can be part of this scope as $b can represent multiple values.


If possible to create a path in which this issue is the first and the most basic and common use case and further in the path end ability to build full statistical analysis and scientific equations it will be great.

**Comment by marvin-hansen** [comment]
Is there any chance arithmetic will be added? 

SQL supports basic operations in queries since forever and I'm quite perplexed to see this one MIA.

**Comment by flyingsilverfin** [comment]
It's on our roadmap now, probably next quarter or the one after. That being said, the reason we haven't prioritised arithmetic in simple `match` queries is because the SQL-like `where` or `select` clauses with operators can be implemented without loss of generality on the application side (with some sacrifice on network overheads). 

That being said, one arithmetic feature that will be brand new and increase the power of TypeDB as a whole would be to include arithmetic/computation in rules, as part of the inference process.

**Comment by flyingsilverfin** [comment]
This has been released with 2.18.0 and implemented in #6732 !
